### PR TITLE
[tween] Add Tween library

### PR DESCRIPTION
**Pull request by sonwh98**
[tween] externs for tween.js 16.3.1

### CODE REVIEW

File: tween/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# cljsjs/three
```
Comment (pandeiro) [inline tween/README.md]: Shouldn't this line be the new library name as well?

**Comment by pandeiro** [comment]
@sonwh98 Hey, what do you think about this externs file I generated at http://www.dotnetwise.com/Code/Externs/ using https://raw.githubusercontent.com/tweenjs/tween.js/master/src/Tween.js ? Could we use this instead?

```
var TWEEN = {
    "getAll": function () {},
    "removeAll": function () {},
    "add": function () {},
    "remove": function () {},
    "update": function () {},
    "Tween": function () {},
    "Easing": {
        "Linear": {
            "None": function () {}
        },
        "Quadratic": {
            "In": function () {},
            "Out": function () {},
            "InOut": function () {}
        },
        "Cubic": {
            "In": function () {},
            "Out": function () {},
            "InOut": function () {}
        },
        "Quartic": {
            "In": function () {},
            "Out": function () {},
            "InOut": function () {}
        },
        "Quintic": {
            "In": function () {},
            "Out": function () {},
            "InOut": function () {}
        },
        "Sinusoidal": {
            "In": function () {},
            "Out": function () {},
            "InOut": function () {}
        },
        "Exponential": {
            "In": function () {},
            "Out": function () {},
            "InOut": function () {}
        },
        "Circular": {
            "In": function () {},
            "Out": function () {},
            "InOut": function () {}
        },
        "Elastic": {
            "In": function () {},
            "Out": function () {},
            "InOut": function () {}
        },
        "Back": {
            "In": function () {},
            "Out": function () {},
            "InOut": function () {}
        },
        "Bounce": {
            "In": function () {},
            "Out": function () {},
            "InOut": function () {}
        }
    },
    "Interpolation": {
        "Linear": function () {},
        "Bezier": function () {},
        "CatmullRom": function () {},
        "Utils": {
            "Linear": function () {},
            "Bernstein": function () {},
            "Factorial": function () {},
            "CatmullRom": function () {}
        }
    }
}
```

File: tween/README.md (hunk 1 above)
Comment (sonwh98) [inline tween/README.md]: good eye! thanks

**Comment by sonwh98** [comment]
@pandeiro awesome! I saw that tool too but didn't know how to use it to generate the extern. Generating extern manually is tedious so I only generated the functions that I needed.

That tool is not complete. It did not include the externs that I was using so it broke my code.  In any case, I will copy the extern you generated with mine.

**Comment by sonwh98** [comment]
I get an error message "Load Failed: https://raw.githubusercontent.com/tweenjs/tween.js/master/src/Tween.js" when using the tool http://www.dotnetwise.com/Code/Externs/. How were you able to get it to work?

**Comment by martinklepsch** [comment]
There is one other externs generator tool that might help: https://github.com/cljsjs/packages/wiki/Creating-Externs#extern-generators

**Comment by dhruvbhatia** [comment]
@sonwh98 Github serves it's raw files as `text/plain` rather than the `text/javascript` MIME type that the generators expect, but you can easily workaround this by replacing `raw.githubusercontent.com` with `raw.githack.com` in the input URL you use.

E.g. try `https://raw.githack.com/tweenjs/tween.js/master/src/Tween.js` in [the generator tool](http://jmmk.github.io/javascript-externs-generator/).

Another alternative is http://rawgit.com/ as pandeiro has used above.

**Comment by sonwh98** [comment]
@dhruvbhatia  great! thanks for clarifying that.
@martinklepsch this pull request is in "extern missing" and "waiting for change". I merged the generated extern with a custom made extern to catch the ones the generator missed. Is there anything else I need to do to move this forward?

**Comment by Deraen** [comment]
Looks good to me.

**Comment by Deraen** [comment]
Deployed

**Pull request merged**